x-pack/plugins/monitoring/init.js

```diff
@@ -21,26 +28,67 @@ import { createCollectorSet } from './server/kibana_monitoring';
  * @param server {Object} HapiJS server instance
  */
 export const init = (monitoringPlugin, server) => {
-  monitoringPlugin.status.yellow('Initializing');
-  const xpackMainPlugin = server.plugins.xpack_main;
+  const config = server.config();
 
-  xpackMainPlugin.status.once('green', async () => {
-    const config = server.config();
+  const collectorSet = new CollectorSet(server);
+  server.expose('collectorSet', collectorSet); // expose the collectorSet service
+  /*
+   * Register collector objects for stats to show up in the APIs
+   */
+  collectorSet.register(getOpsStatsCollector(server));
+  collectorSet.register(getKibanaUsageCollector(server));
+  collectorSet.register(getSettingsCollector(server));
+
+  /*
+   * Instantiate and start the internal background task that calls collector
+   * fetch methods and uploads to the ES monitoring bulk endpoint
+   */
+  const xpackMainPlugin = server.plugins.xpack_main;
+  xpackMainPlugin.status.once('green', async () => { // first time xpack_main turns green
+    /*
+     * End-user-facing services
+     */
     const uiEnabled = config.get('xpack.monitoring.ui.enabled');
 
     if (uiEnabled) {
       await instantiateClient(server); // Instantiate the dedicated ES client
       await initMonitoringXpackInfo(server); // Route handlers depend on this for xpackInfo
       await requireUIRoutes(server);
     }
+  });
 
-    if (config.get('xpack.monitoring.kibana.collection.enabled')) {
-      const collectorSet = createCollectorSet(monitoringPlugin.kbnServer, server); // instantiate an object for collecting/sending metrics and usage stats
-      server.expose('collectorSet', collectorSet); // expose the collector set object on the server. other plugins will call statsCollectors.register(collector) to define their own collection
-    }
+  const kibanaCollectionEnabled = config.get('xpack.monitoring.kibana.collection.enabled');
+  const { info: xpackMainInfo } = xpackMainPlugin;
 
-    monitoringPlugin.status.green('Ready');
-  });
+  if (kibanaCollectionEnabled && xpackMainInfo) {
+    const bulkUploader = initBulkUploader(monitoringPlugin.kbnServer, server, xpackMainInfo);
+
+    xpackMainPlugin.status.on('green', async () => { // any time xpack_main turns green
+      /*
+       * Bulk uploading of Kibana stats
+       */
+
+      bulkUploader.start(collectorSet);
+    });
+
+    xpackMainPlugin.status.on('red', () => { // any time xpack_main turns red
+      bulkUploader.stop();
+    });
+  }
+
+  if (!kibanaCollectionEnabled) {
+    server.log(
+      ['info', LOGGING_TAG, KIBANA_MONITORING_LOGGING_TAG],
+      'Internal collection for Kibana monitoring will is disabled per configuration.'
+    );
+  }
+
+  if (!xpackMainInfo) {
```

Comment:
This should probably be an `else` attached to the previous two `if`s to avoid double-logging in a rare situation when they told us they didn't want to do it anyway.